src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -474,16 +471,18 @@ class NetworkOPsImp final
     void onStop () override
     {
         mAcquiringLedger.reset();
-        m_heartbeatTimer.cancel();
-        m_clusterTimer.cancel();
+
+        if (heartbeatTimer_)
+            heartbeatTimer_->cancel();
+        if (clusterTimer_)
+            clusterTimer_->cancel();
 
         stopped ();
```

Comment:
You can post to the io_service thread.

`heartbeatTimer_->get_io_service().post([=] { heartbeatTimer_->cancel(); });`